tools/telemetry-generator/src/handlers/testPassRate.ts

```diff
@@ -0,0 +1,41 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+/**
+ * This handler uses the timeline and metrics REST APIs to retrieve the test pass rate of the pipeline:
+ * Ex. https://dev.azure.com/fluidframework/internal/_apis/build/builds/<buildId>/timeline?api-version=7.1-preview.1
+ * Ex. https://vstmr.dev.azure.com/fluidframework/internal/_apis/testresults/metrics?pipelineId=<buildId>&stageName=<stageName>&api-version=7.1-preview.1
+ * @param fileData - A JSON object obtained by calling JSON.parse() on the contents of a file.
+ * @param logger - An ITelemetryBufferedLogger. Call its send() method to write the output telemetry events.
+ */
+module.exports = function handler(fileData, logger) {
+	if (fileData.resultSummary == null ) {
+		console.log(`Could not locate test result info.`);
+		return;
+	}
+	if (process.env.BUILD_ID !== undefined) {
+		console.log("BUILD_ID", process.env.BUILD_ID);
+	} else {
+		console.log("BUILD_ID not defined.");
+	}
+	const resultSummary = fileData.resultSummary.resultSummaryByRunState.Completed;
+	console.log(resultSummary);
+
+	const passedTests: number = resultSummary.aggregatedResultDetailsByOutcome.Passed?.count ?? 0;
+	const failedTests: number = resultSummary.aggregatedResultDetailsByOutcome.Failed?.count ?? 0;
+	const totalTests = passedTests + failedTests;
+	const passRate = totalTests !== 0 ? passedTests / totalTests : 0;
+	console.log(passRate);
+	logger.send({
+		category: "performance",
+		eventName: "TestPassRate",
+		benchmarkType: "PipelineInfo",
+		stageName: fileData.currentContext.stageReference.stageName,
+		result: passRate,
```

Comment:
I think it's also worth it to include the total test count, and absolute number of passed and failed ones, like we had before. Let's restore them. If we ever see a testPassRate of 100% but only 1 test executed, something is wrong.
```suggestion
		testPassRate: passRate,
```